recipe-client-addon/test/unit/test_PreferenceExperiments.js

```diff
@@ -594,20 +611,21 @@ add_task(withMockExperiments(async function testInitTelemetryExpired(experiments
 
 // Experiments should end if the preference has been changed when init() is called
 add_task(withMockExperiments(withMockPreferences(async function testInitChanges(experiments, mockPreferences) {
+  const stopStub = sinon.stub(PreferenceExperiments, "stop");
   experiments["test"] = experimentFactory({
     name: "test",
     preferenceName: "fake.preference",
     preferenceValue: "experiment value",
   });
   mockPreferences.set("fake.preference", "changed value");
   await PreferenceExperiments.init();
-  ok(experiments["test"].expired, "Experiment expires because value changed");
+  ok(stopStub.calledWith("test"), "Experiment is stopped because value changed");
```

Comment:
`calledWith` only asserts about the arguments you pass. so

```js
const foo = sinon.stub();
foo(1, 2, 3);
ok(foo.calledWith(1));
```

would pass. The behavior you're thinking of is a part of `calledWithExact`. Which we should probably be using.